pkg/build/api/v1beta3/types.go

```diff
@@ -167,6 +172,31 @@ type BuildSource struct {
 	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
 }
 
+// ImageSource describes an image that is used as source for the build
+type ImageSource struct {
+	// From is a reference to an ImageStream tag or a DockerImage to
+	// copy source from
+	From kapi.ObjectReference `json:"from" description:"image reference"`
+
+	// Paths is a list of source and destination paths to copy from the image
+	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
+
+	// PullSecret is a reference to a secret to be used to pull the image from
+	// a registry
+	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"pull secret for source image"`
```

Comment:
If source secrets is going to move add that as a todo and make sure
your new name here is consistent with what it will be eventually.
Also note in the description that it is an override.